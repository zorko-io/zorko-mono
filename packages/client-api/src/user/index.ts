export * from './user.api.client';
export * from './user.axios.api.client';
