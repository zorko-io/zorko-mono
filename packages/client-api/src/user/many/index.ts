export * from './user.many.api.client'
export * from './user.axios.many.api.client'
