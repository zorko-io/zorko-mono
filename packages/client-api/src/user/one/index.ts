export * from './user.one.api.client'
export * from './user.axios.one.api.client'
