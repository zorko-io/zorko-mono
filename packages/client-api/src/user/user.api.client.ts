import { AuthApiClient } from '../auth';
import { RemoteUserApi } from '@zorko/remote-api';

export interface UserApiClient extends AuthApiClient, RemoteUserApi {}
