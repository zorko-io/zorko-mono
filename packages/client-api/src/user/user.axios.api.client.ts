import { AxiosResponse } from 'axios';
import { AuthAxiosApiClient } from '../auth';
import { UserDto } from '@zorko/dto';
import { UserApiClient } from './user.api.client';
import {
  CreateUserCollectionParams,
  CreateUserParams, DeleteUserCollectionParams, DeleteUserParams,
  ReadUserCollectionParams,
  ReadUserParams,
  UpdateUserCollectionParams, UpdateUserParams,
} from '@zorko/remote-api';

export class UserAxiosApiClient extends AuthAxiosApiClient implements UserApiClient  {

  async findMany(readParams?: ReadUserCollectionParams): Promise<UpdateUserCollectionParams> {
    const response: AxiosResponse<UpdateUserCollectionParams> = await this.http.get('/users');
    return response.data
  }

  async createOne(user: CreateUserParams): Promise<string> {
    const response: AxiosResponse<string> = await this.http.post('/users', user);
    return response.data;
  }

  async findOne(readParams: ReadUserParams): Promise<UpdateUserParams> {
    const { id }  = readParams;
    const response: AxiosResponse<UserDto> = await this.http.get(`/users/${id}`);
    return response.data;
  }

  async createMany(createParams: CreateUserCollectionParams): Promise<string> {
    return '123'
  }

  async removeMany(deleteParams: DeleteUserCollectionParams): Promise<string> {
    const response: AxiosResponse<string> = await this.http.delete('/users');
    return response.data
  }

  async removeOne(deleteParams: DeleteUserParams): Promise<void> {
    const { id } = deleteParams;
    await this.http.delete(`/users/${id}`);
  }

  updateMany(updateParams: UpdateUserCollectionParams): Promise<ReadUserCollectionParams> {
    return undefined;
  }

  updateOne(updateParams: UpdateUserParams): Promise<ReadUserParams> {
    return undefined;
  }
}

