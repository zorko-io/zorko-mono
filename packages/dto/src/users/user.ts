import { RolesEnum } from '../roles';

export interface User {
  id?: any;
  login?: string;
  email: string;
  password?: string;
  hashPassword?: string;

  roles?: RolesEnum[];
}
