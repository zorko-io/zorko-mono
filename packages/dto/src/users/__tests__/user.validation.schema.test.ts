import { User } from '../user';
import { userValidationSchema } from '../user.validation.schema';
import { userFakeGenerator } from '../user.fake.generator';

describe('userValidationSchema', () => {
  let schema;
  let data: User;

  beforeEach(() => {
    schema = userValidationSchema();
  });

  it('passes minimum validation with `email`', () => {
    data = userFakeGenerator.getMinimalRandomValidUser();
    expect(schema.isValidSync(data)).toBeTruthy();
  });

  it('passes max validation', () => {
    data = userFakeGenerator.getRandomValidUser();
    expect(schema.isValidSync(data)).toBeTruthy();
  });

  it('passes `hashPassword` instead of real password', () => {
    data = userFakeGenerator.getRandomValidUserOnlyWithHashPassword();
    expect(schema.isValidSync(data)).toBeTruthy();
  });

  it('fails with wrong email', () => {
    data = userFakeGenerator.getRandomUserWithWrongEmail();
    expect(() => {
      schema.validateSync(data);
    }).toThrowErrorMatchingSnapshot();
  });

  it('fails without login but with `hashPassword`', () => {
    data = userFakeGenerator.getRandomUserWithoutLogin();
    expect(() => {
      schema.validateSync(data);
    }).toThrowErrorMatchingSnapshot();
  });

});
