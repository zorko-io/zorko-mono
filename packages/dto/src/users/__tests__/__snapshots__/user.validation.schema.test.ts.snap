// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userValidationSchema fails with wrong email 1`] = `"email must be a valid email"`;

exports[`userValidationSchema fails without login but with \`hashPassword\` 1`] = `"login is a required field"`;
